### PR TITLE
[PECO-1048] Add example for parameterized queries

### PR DESCRIPTION
**Pull request by nithinkdb**
We don't currently have an example for parameterized queries, we should add one.

### CODE REVIEW

File: examples/parameters/main.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	var res string
+	err1 := db.QueryRowContext(ctx, `SELECT
+	:p_bool AS col_bool,
+	:p_int AS col_int,
+	:p_double AS col_double,
+	:p_floatfloat AS col_float`,
+		dbsql.DBSqlParam{Name: "p_bool", Value: true},
+		dbsql.DBSqlParam{Name: "p_int", Value: int(1234)},
+		dbsql.DBSqlParam{Name: "p_double", Type: dbsql.Double, Value: 3.14},
+		dbsql.DBSqlParam{Name: "p_float", Type: dbsql.Float, Value: 3.14}).Scan(&res)
```
Comment (rcypher-databricks) [inline examples/parameters/main.go:51]: I think we need to have a param that requires the use of DBSqlParam.Type.  Maybe a date where the passed value is a string. And have a comment about how if Type is not specified we try to infer it.

**Review by rcypher-databricks — COMMENTED** [review]
As part of this PR please add a section in doc.go explaining the feature and how to use it.
Including things like what happens when passing raw values or sql.NamedValue as parameters.

**Comment by nithinkdb** [comment]
> As part of this PR please add a section in doc.go explaining the feature and how to use it. Including things like what happens when passing raw values or sql.NamedValue as parameters.

Adding in a future PR.

File: examples/parameters/main.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	var p_double float64
+	var p_float float32
+	var p_date string
+	err1 := db.QueryRowContext(ctx, `SELECT
+	:p_bool AS col_bool,
+	:p_int AS col_int,
+	:p_double AS col_double,
+	:p_float AS col_float,
+	:p_date AS col_date`,
+		dbsql.DBSqlParam{Name: "p_bool", Value: true},
```
Comment (andrefurlan-db) [inline examples/parameters/main.go:53]: nothing to do with this PR, but why is this called DBSqlParam? Please, please standardize the names we expose as public interfaces. @rcypher-databricks what do you think?

Comment (andrefurlan-db) [reply examples/parameters/main.go:53]: we should just call this Param. As it is already scoped to dbsql

Comment (rcypher-databricks) [reply examples/parameters/main.go:53]: @nithinkdb I though we talked about dropping DBSql and just calling it Parameter.

Comment (nithinkdb) [reply examples/parameters/main.go:53]: Sorry, forgot about this. Will do.

**Review by rcypher-databricks — APPROVED** [review]

**Pull request merged**